data/ignition/systemd/units/agent.service

```diff
@@ -0,0 +1,25 @@
+[Service]
+Type=simple
+Restart=always
+RestartSec=3
+StartLimitInterval=0
+Environment=HTTP_PROXY=
+Environment=http_proxy=
+Environment=HTTPS_PROXY=
+Environment=https_proxy=
+Environment=NO_PROXY=
+Environment=no_proxy=
+# TODO: If AUTH_TYPE != none, then PULL_SECRET_TOKEN needs to be updated
+# https://github.com/openshift/assisted-service/blob/master/internal/ignition/ignition.go#L1381
+Environment=PULL_SECRET_TOKEN={{.PullSecretToken}}
+TimeoutStartSec=180
+ExecStartPre=/usr/local/bin/agent-fix-bz1964591 quay.io/edge-infrastructure/assisted-installer-agent:latest
+ExecStartPre=podman run --privileged --rm -v /usr/local/bin:/hostbin quay.io/edge-infrastructure/assisted-installer-agent:latest cp /usr/bin/agent /hostbin
+ExecStart=/usr/local/bin/agent --url {{.ServiceBaseURL}} --infra-env-id {{.infraEnvId}} --agent-version quay.io/edge-infrastructure/assisted-installer-agent:latest --insecure=true
+
+[Unit]
+Wants=network-online.target
+After=network-online.target
```

Comment:
Worth putting After assisted-service.service as well?

Reply:
If only one host will have assisted-service.service, then the agents shouldn't wait on it.

Reply:
In systemd, `After` only has an effect if the other service is enabled, so it won't wait for a service that isn't started.
With multiple hosts there's no way to guarantee that the agents on the non-Node0 hosts start after assisted-service, so maybe it doesn't matter. But there might be some minor benefit like slightly cleaner logs.